File: GeneratedEntityReaderClasses/Components/UIIconComponent.cs
namespace GeneratedNoitaClasses;
public class UIIconComponent : NoitaComponentBase,  iNoitaType<UIIconComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString icon_sprite_file { get; set; } = new();
	public NoitaString name { get; set; } = new();
	public NoitaString description { get; set; } = new();
	public NoitaBool display_above_head { get; set; } = new();
	public NoitaBool display_in_hud { get; set; } = new();
	public NoitaBool is_perk { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		icon_sprite_file.Read(s);
		name.Read(s);
		description.Read(s);
		display_above_head.Read(s);
		display_in_hud.Read(s);
		is_perk.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		icon_sprite_file.Write(s);
		name.Write(s);
		description.Write(s);
		display_above_head.Write(s);
		display_in_hud.Write(s);
		is_perk.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/UIIconComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString

from NoitaStream import NoitaStream
@dataclass
class UIIconComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    icon_sprite_file:NoitaString = field(default_factory=NoitaString)
    name:NoitaString = field(default_factory=NoitaString)
    description:NoitaString = field(default_factory=NoitaString)
    display_above_head:NoitaBool = field(default_factory=NoitaBool)
    display_in_hud:NoitaBool = field(default_factory=NoitaBool)
    is_perk:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.icon_sprite_file.Read(s)
        self.name.Read(s)
        self.description.Read(s)
        self.display_above_head.Read(s)
        self.display_in_hud.Read(s)
        self.is_perk.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.icon_sprite_file.Write(s)
        self.name.Write(s)
        self.description.Write(s)
        self.display_above_head.Write(s)
        self.display_in_hud.Write(s)
        self.is_perk.Write(s)

